docs/configuration.md

```diff
@@ -44,6 +44,8 @@ but overflows could happen in case of slow provisioners and / or higher number o
 concurrent reconciles. For such reasons, it is highly recommended to keep
 BMO_CONCURRENCY value lower than the requested PROVISIONING_LIMIT. Default is 20.
 
+`IRONIC_EXTERNAL_URL` -- This is the URL where Ironic will find the image for the node.
+
```

Comment:
Trying to understand this, is it similar to virtual media? Can you please some more information on the use case where this would be suitable?

Reply:
So, some background (I agree it should be here in the docs): on a dual stack setup, Ironic has no way to understand which IP version to use for telling a BMC the address of its HTTP server. So the idea here is: default to v4 and use v6 when BMC itself is v6.

Reply:
I updated the docs but it feels cumbersome. Suggestions for improvement are appreciated.

Reply:
It's pretty awkward that the v4 URL and the v6 URL are set through different mechanisms (the former is set in the ironic config iirc?).

Reply:
@zaneb ironic doesn't have the concept of v4/v6 for the BMC, in case people configure the
`[deploy]/external_http_url and [deploy]/http_url` with v6 information this would be the default it would use, now we added the option to override the url at node level so the operator can specify what they want for each node. In our use case for dual-stack we want to use this for v6 only (so BMO would identify the IPversion of the BMC and set if there is the value from the config)

Reply:
So if the URLs are settable on a per-host basis, it sounds like one option would be to pass both the v4 and v6 URLs to the BMO and have it always set the right one for each host. (Obviously another would be to modify ironic to accept both URLs in the config and either select one automatically or allow the BMO to specify through the API which one to use.)

Having two completely different data flows for the v4 and v6 URLs in a dual-stack cluster is a serious maintenance burden for whoever has to deploy metal³. Sometimes such things are necessary in the short term, but ideally they should come with at least a plan for unwinding them.